crates/kitsune_p2p/bootstrap/src/store.rs

```diff
@@ -1,15 +1,52 @@
 use std::{collections::HashMap, sync::Arc};
 
 use kitsune_p2p_types::{
-    agent_info::AgentInfoSigned,
-    bin_types::{KitsuneAgent, KitsuneSpace},
+    bin_types::{KitsuneAgent, KitsuneSignature, KitsuneSpace},
     bootstrap::RandomQuery,
-    codec::rmp_encode,
 };
 use parking_lot::RwLock;
 use rand::seq::IteratorRandom;
 
-type AgentMap = HashMap<Arc<KitsuneAgent>, AgentInfoSigned>;
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq, Clone))]
+pub(crate) struct StoreEntry {
+    pub encoded: Vec<u8>,
+    pub signature: Arc<KitsuneSignature>,
+    pub space: Arc<KitsuneSpace>,
+    pub agent: Arc<KitsuneAgent>,
+    pub signed_at_ms: u64,
+    pub expires_at_ms: u64,
+}
+
+impl StoreEntry {
+    pub fn parse(encoded: Vec<u8>) -> Result<Self, std::io::Error> {
+        let mut bytes: &[u8] = &encoded;
+        let kitsune_p2p_types::agent_info::agent_info_helper::AgentInfoSignedEncode {
+            agent,
+            signature,
+            agent_info,
+        } = kitsune_p2p_types::codec::rmp_decode(&mut bytes)?;
+
+        let mut bytes: &[u8] = &agent_info;
+        let info: kitsune_p2p_types::agent_info::agent_info_helper::AgentInfoEncode =
+            kitsune_p2p_types::codec::rmp_decode(&mut bytes)?;
+
+        if agent != info.agent {
+            return Err(std::io::Error::other("agent mismatch"));
+        }
```

Comment:
This error could be better, maybe give some info about which agents mismatched?

Reply:
```suggestion
        if agent != info.agent {
            return Err(std::io::Error::other("signed inner agent does not match unsigned outer agent"));
        }
```